src/components/auth-module/register-view/register-view.js

```diff
@@ -0,0 +1,209 @@
+
+  /***************************
+   *          METHODS
+   ***************************/
+  handleChange(e){
+    let fields = this.state.fields;
+    const field = e.target.name;
+    fields[field] = e.target.value;
+    this.setState({
+      fields,
+      errors: {},
+      serverError: ""
+    })
+  }
+
+  handleSubmit = (e)=> {
+    e.preventDefault();
+    const formValidationFeedback = this.handleFormValidation(this.state.fields);
+    if(formValidationFeedback.isFormValid){
+      // API call for register request
+      this.setState({showErrorMessage: true});
+    }else{
+      this.setState({errors:formValidationFeedback.errors});
+    }
+  }
+
+  handleFormValidation(fields){
+    let errors = {};
+    let formIsValid = true;
+    let result = {};
+
+    if(fields['firstName'] === ''){
+      formIsValid = false;
+      errors["firstName"] = "*This field is required.";
+    }
+    if(fields['email'] === ''){
+      formIsValid = false;
+      errors["email"] = "*This field is required.";
+    } else if(typeof fields["email"] !== "undefined"){
+      if (!isValidEmail(fields["email"])) {
+        formIsValid = false;
+        errors["email"] = "Please provide valid email id.";
+      }
+    }
+    if(fields['phoneNumber'] === ''){
+      formIsValid = false;
+      errors["phoneNumber"] = "*This field is required.";
+    }
+
+    result['errors'] = errors;
+    result['isFormValid'] = formIsValid;
+    return result;   
+  }
+
+  handleClearError() {
+    this.setState({showErrorMessage: false, showSuccessMessage: false});
+  }
+
+  /***************************
+   *           VIEWS
+   ***************************/
+  getFieldErrorView(error){
+    return(
+      <ErrorMessageView error={error} />
+    )
+  }
+
+  /***************************
+   *         LIFECYCLE
+   ***************************/
+  render(){
+    const { t } = this.props;
+    const { errors, fields, serverError } = this.state;
+
+    return(
+      <div className="auth-form-view">
+        <div className="left-section">
+          <div className="brand-name">{t('common.brandName')}</div>
+          <div className="left-section-marker"></div>
+        </div>
+        <div className="right-section">
+          <div className="register-view-form-wrapper">
+            <div className="heading">{t('auth.registerHeading')}</div>
+            {
+              this.state.showSuccessMessage &&
+              <RegisterMessageView
+              registerSuccess={this.state.showSuccessMessage}
+              handleClearError={()=>this.handleClearError()}/>
+            }
+            {
+              this.state.showErrorMessage &&
+              <RegisterMessageView
+              registerError={this.state.showErrorMessage}
+              handleClearError={()=>this.handleClearError()}/>
+            }
+            {
+              !this.state.showSuccessMessage && !this.state.showErrorMessage &&
+              <Form className="register-form" onSubmit= {this.handleSubmit}>
```

Comment:
Better to move this form code to separate function and return jsx from there. With that your code should look something like below
`!showSuccessMessage && !showErrorMessage && this.getRegisterVIew()`